sinonjs/referee
# Upgrade samsam and dependency maintenance

#### Purpose (TL;DR) - mandatory

Referee referred to an outdated version of `@sinonjs/samsam` causing the library to exist twice in the dependency tree when depending on `sinon` as well (e.g. `@sinonjs/referee-sinon`).

Since this will be released as a breaking change, I also made the following changes:

- Removed support for node 12, added node 18.
- Upgrades `devDependencies` and applies new prettier formatting.

#### How to verify - mandatory

1. Check out this branch
2. `npm install`
3. `npm run lint`
4. `npm run prettier:check`
5. `npm test`

#### Checklist for author
